app/src/main/java/org/mozilla/fenix/customtabs/WebAppSiteControlsBuilder.kt

```diff
@@ -0,0 +1,44 @@
+package org.mozilla.fenix.customtabs
+
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import androidx.core.app.NotificationCompat
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.feature.pwa.feature.SiteControlsBuilder
+import mozilla.components.feature.session.SessionUseCases
+import org.mozilla.fenix.R
+
+class WebAppSiteControlsBuilder(
+    private val sessionManager: SessionManager,
+    reloadUrlUseCase: SessionUseCases.ReloadUrlUseCase,
+    private val sessionId: String,
+    private val manifest: WebAppManifest
+) : SiteControlsBuilder {
+
+    private val inner = SiteControlsBuilder.CopyAndRefresh(reloadUrlUseCase)
+
+    override fun buildNotification(
+        context: Context,
+        builder: NotificationCompat.Builder,
+        channelId: String
+    ) {
+        inner.buildNotification(context, builder, channelId)
+
+        val session = sessionManager.findSessionById(sessionId)
+        val isPrivateSession = session?.private ?: false
+
+        if (!isPrivateSession) {
+            return;
```

Comment:
Semicolon not necessary here, might also be nice to just have this be in one line to match the style in the rest of the app:

`if (!isPrivateSession) { return }`

Reply:
In fact, I may even collapse the code above into:

`val isPrivateSession = sessionManager.findSessionById(sessionId)?.private ?: false`

since we're not using the `session` variable